core/processor/src/main/java/io/quarkus/annotation/processor/generate_doc/ConfigDocItemFinder.java

```diff
@@ -332,6 +330,10 @@ private List<ConfigDocItem> recursivelyFindConfigItems(Element element, String r
                                 acceptedValues = extractEnumValues(realTypeMirror, useHyphenateEnumValue,
                                         clazz.getQualifiedName().toString());
                                 configDocKey.setEnum(true);
+                            } else {
```

Comment:
I prefer to keep the `else` as is rather than group in `else if` to have a proper separate case when the config type is not an enum.